src/query/frontend/ast/ast.hpp

```diff
@@ -2069,6 +2069,27 @@ class Query : public memgraph::query::Tree, public utils::Visitable<QueryVisitor
   friend class AstStorage;
 };
 
+struct IndexHint {
+  static const utils::TypeInfo kType;
+  const utils::TypeInfo &GetTypeInfo() const { return kType; }
+
+  enum class IndexType { LABEL, LABEL_PROPERTY };
+
+  memgraph::query::IndexHint::IndexType index_type_;
+  memgraph::query::LabelIx label_;
+  std::optional<memgraph::query::PropertyIx> property_ = std::nullopt;
+
+  IndexHint Clone(AstStorage *storage) const {
+    IndexHint object;
+    object.index_type_ = index_type_;
+    object.label_ = storage->GetLabelIx(label_.name);
```

Comment:
why not?
`object.label_ = label_;`

Reply:
I followed the practice used in existing code (see `IndexQuery` and `Constraint`).